### PR TITLE
[1LP][RFR] New tenant quota tests for cloud

### PR DESCRIPTION
**Pull request by vatsalparekh**
Purpose or Intent
=================
New tenant quota test for cloud
{{ pytest: -vvvv cfme/tests/cloud/test_tenant_quota.py }}

### CODE REVIEW

**Comment by dajoRH** [comment]
Lint report for commit 71b5c276fba4652ecf6d204bc50d1a9590546c0a:

`cfme/tests/cloud/test_tenant_quota.py`:
- :bangbang: Line 1:1: **E902** *TokenError: EOF in multi-line statement*
- :bangbang: Line 117:56: **E999** *SyntaxError: invalid syntax*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 1840c66fc9eee48a17ed54d5fa889af685af37ea:

`cfme/tests/cloud/test_tenant_quota.py`:
- :bangbang: Line 1:1: **E902** *TokenError: EOF in multi-line statement*
- :bangbang: Line 116:56: **E999** *SyntaxError: invalid syntax*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

File: cfme/tests/cloud/test_tenant_quota.py [hunk 1]
```diff
@@ -0,0 +1,141 @@
+# -*- coding: utf-8 -*-
+import fauxfactory
+import pytest
+from cfme import test_requirements
```
Comment (quarckster) [inline cfme/tests/cloud/test_tenant_quota.py]: **REQUIRED**
Please put one blank between external and internal imports.

File: cfme/tests/cloud/test_tenant_quota.py [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+    request_description = 'Provision from [{}] to [{}{}]'.format(template_name, vm_name, extra_msg)
+    provision_request = appliance.collections.requests.instantiate(request_description)
+    if approve:
+        provision_request.approve_request(method='ui', reason="Approved")
+    provision_request.wait_for_request(method='ui')
+    assert provision_request.row.reason.text == "Quota Exceeded"
+
+    @request.addfinalizer
+    def delete():
+        provision_request.remove_request()
```
Comment (quarckster) [inline cfme/tests/cloud/test_tenant_quota.py]: **OPTIONAL**
```python
request.addfinalizer(provision_request.remove_request)
```

Comment (vatsalparekh) [reply cfme/tests/cloud/test_tenant_quota.py]: added.

File: cfme/tests/cloud/test_tenant_quota.py (hunk 1 above)
Comment (vatsalparekh) [reply cfme/tests/cloud/test_tenant_quota.py]: added.

File: cfme/tests/cloud/test_tenant_quota.py [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+import fauxfactory
+import pytest
+
+from cfme import test_requirements
+from cfme.cloud.instance import Instance
+from cfme.cloud.provider.openstack import OpenStackProvider
+from cfme.services.catalogs.catalog_item import CatalogItem
+from cfme.services.service_catalogs import ServiceCatalogs
+from cfme.utils.generators import random_vm_name
+from widgetastic.utils import Version, VersionPick
```
Comment (quarckster) [inline cfme/tests/cloud/test_tenant_quota.py]: **REQUIRED**
`from widgetastic.utils import Version, VersionPick` should be moved to external import block

File: cfme/tests/cloud/test_tenant_quota.py [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+
+pytestmark = [
+    test_requirements.quota,
+    pytest.mark.provider([OpenStackProvider], required_fields=[['provisioning', 'image']],
```
[the remark this answers is not in the document]
Comment (vatsalparekh) [reply cfme/tests/cloud/test_tenant_quota.py]: No, they are planned to come in later PR.

File: cfme/tests/cloud/test_tenant_quota.py [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+    return {
+        "catalog": {'vm_name': vm_name},
+        "environment": {'automatic_placement': True},
+    }
+
+
+@pytest.fixture
+def set_roottenant_quota(request, roottenant):
+    field, value = request.param
+    roottenant.set_quota(**{'{}_cb'.format(field): True, field: value})
```
[the remark this answers is not in the document]
Comment (vatsalparekh) [reply cfme/tests/cloud/test_tenant_quota.py]: :+1:

File: cfme/tests/cloud/test_tenant_quota.py [hunk 6]
```diff
@@ -0,0 +1,141 @@
...
+    return random_vm_name(context='quota')
+
+
+@pytest.fixture
+def template_name(provisioning):
+    return provisioning["image"]["name"]
+
+
+@pytest.fixture(scope="module")
+def roottenant(appliance):
```
[the remark this answers is not in the document]
Comment (vatsalparekh) [reply cfme/tests/cloud/test_tenant_quota.py]: I feel like keeping it roottenant, have kept it in all other tenant tests

File: cfme/tests/cloud/test_tenant_quota.py (hunk 3 above)
Comment (vatsalparekh) [reply cfme/tests/cloud/test_tenant_quota.py]: Moved

**Comment by dajoRH** [comment]
Lint report for commit 8e55c096a0d382c0f13038478f003ee2cf526541:

`cfme/tests/cloud/test_tenant_quota.py`:
- :bangbang: Line 91:28: **E127** *continuation line over-indented for visual indent*
- :bangbang: Line 91:101: **E501** *line too long (119 > 100 characters)*
- :bangbang: Line 128:101: **E501** *line too long (117 > 100 characters)*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

**Pull request merged**